src/Domain/Wallet.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using dotnetstandard_bip39;
+using Erdcsharp.Cryptography;
+using Erdcsharp.Domain.Helper;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Crypto.Signers;
+
+namespace Erdcsharp.Domain
+{
+    public class Wallet
+    {
+        private readonly        byte[]                   _privateKey;
+        private readonly        byte[]                   _publicKey;
+        private const           string                   HdPrefix = "m/44'/508'/0'/0'";
+        private static readonly RNGCryptoServiceProvider RngCsp   = new RNGCryptoServiceProvider();
+
+        public Wallet(string privateKeyHex)
+            : this(Converter.FromHexString(privateKeyHex))
+        {
+        }
+
+        /// <summary>
+        /// Get the account wallet
+        /// </summary>
+        /// <returns>Account</returns>
+        public Account GetAccount()
+        {
+            return new Account(Address.FromBytes(_publicKey));
+        }
+
+        /// <summary>
+        /// Build a wallet
+        /// </summary>
+        /// <param name="privateKey">The private key</param>
+        public Wallet(byte[] privateKey)
+        {
+            var privateKeyParameters = new Ed25519PrivateKeyParameters(privateKey, 0);
+            var publicKeyParameters  = privateKeyParameters.GeneratePublicKey();
+            _publicKey  = publicKeyParameters.GetEncoded();
+            _privateKey = privateKey;
+        }
+
+        /// <summary>
+        /// Derive a wallet from Mnemonic phrase
+        /// </summary>
+        /// <param name="mnemonic">The mnemonic phrase</param>
+        /// <param name="accountIndex">The account index, default 0</param>
+        /// <returns>Wallet</returns>
+        public static Wallet DeriveFromMnemonic(string mnemonic, int accountIndex = 0)
```

Comment:
Please add more tests for this function (it isn't tested per-se, currently you just call it from `TestWalletFactory`). You can use the well-known accounts etc.

https://github.com/ElrondNetwork/elrond-sdk/tree/master/erdpy/testnet/wallets/users